pkg/controller/external_secrets/deployments.go

```diff
@@ -390,3 +394,128 @@ func updateBitwardenServerContainerSpec(deployment *appsv1.Deployment, image str
 		}
 	}
 }
+
+// updateProxyConfiguration applies all proxy-related configuration to the deployment.
+func (r *Reconciler) updateProxyConfiguration(deployment *appsv1.Deployment, esc *operatorv1alpha1.ExternalSecretsConfig) error {
+	if err := r.updateProxyEnvironmentVariables(deployment, esc); err != nil {
+		return fmt.Errorf("failed to update proxy environment variables: %w", err)
+	}
+	if err := r.updateTrustedCABundleVolumes(deployment, esc); err != nil {
+		return fmt.Errorf("failed to update trusted CA bundle volumes: %w", err)
+	}
+
+	return nil
+}
+
+// updateProxyEnvironmentVariables sets proxy environment variables on all containers in the deployment.
+func (r *Reconciler) updateProxyEnvironmentVariables(deployment *appsv1.Deployment, esc *operatorv1alpha1.ExternalSecretsConfig) error {
+	proxyConfig := r.getProxyConfiguration(esc)
+	if proxyConfig == nil {
+		return nil
+	}
+
+	// Apply proxy environment variables to all containers
+	for i := range deployment.Spec.Template.Spec.Containers {
+		container := &deployment.Spec.Template.Spec.Containers[i]
+		r.setProxyEnvVars(container, proxyConfig)
+	}
+
+	return nil
+}
+
+// setProxyEnvVars sets proxy environment variables on a container.
+func (r *Reconciler) setProxyEnvVars(container *corev1.Container, proxyConfig *operatorv1alpha1.ProxyConfig) {
+	if proxyConfig == nil {
```

Comment:
We should for proxy ENV vars in operator pod set by OLM, when cluster-wide egress proxy is configured.

Reply:
Sorry, I could not understand it correctly. 

We are doing the necessary checks in `getProxyConfiguration` function and feeding the value in `proxyConfig` before calling`setProxyEnvVars`. 

If `proxyConfig` is non-nil value here, then we will set the env vars in the operand pods. 

In case we need any modification to this logic, let me know.

Reply:
My suggestion was to handle [this](https://github.com/openshift/external-secrets-operator/pull/62/files#diff-805d3924ab8e1b0211df38fed8eae18cdd20d55348ec73025cdacb406ec7e726R139-R150) case.

Reply:
Okay, yeah we are already considering OLM Env vars when cluster-wide proxy is enabled.